aws/resource_aws_instance.go

```diff
@@ -893,6 +893,10 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 				groups = append(groups, aws.String(v.(string)))
 			}
 		}
+
+		if len(groups) < 1 {
+			return fmt.Errorf("At least one security group needs to be attached!")
```

Comment:
Since it is true only for VPC-based instances, what do you think of enhancing this message with: `VPC-based instances require at least one security group to be attached`?

Source: 
> [EC2-VPC] Changes the security groups of the instance. You must specify at least one security group, even if it's just the default security group for the VPC. You must specify the security group ID, not the security group name.

Reply:
Does this not apply in EC2 Classic? I haven't tested there and haven't found anything relevant in the AWS docs. If it does only apply to VPC security groups then I think we can make the message more explicit.

Reply:
According to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifyInstanceAttribute.html#API_ModifyInstanceAttribute_RequestParameters under `GroupId.N` you will found the extract I wrote above. Apart from that, can't really tell 🤷‍♂️ 

If it's also applicable to EC2-classic, then we would need to add `MaxItems: 1` to the `vpc_security_group_ids` attribute, which would handle it in all cases.

Can you check it against EC2-classic?

Reply:
I'm unable to test in EC2-classic since my account does not support it. I can only create instances in a VPC. I'll update the error message though.

Reply:
Actually, I was able to test in EC2-classic and after creating an instance security groups cannot be changed. Trying to set vpc_security_groups_ids = [] results in the following error...
```
* aws_instance.web: Failed to update vpc_security_group_ids on "i-0f1be7ea20d091c29",
which does not contain a primary network interface
```

Reply:
Is it possible to *create* an instance without security groups?

Reply:
I do not think so, I had not specified one in the template and on creation it had attached our default VPC security group.